sw/device/silicon_creator/lib/cert/cert.c

```diff
@@ -0,0 +1,220 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/cert/cert.h"
+
+#include <stdint.h>
+
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/status.h"
+#include "sw/device/lib/testing/json/provisioning_data.h"
+#include "sw/device/silicon_creator/lib/attestation.h"
+#include "sw/device/silicon_creator/lib/attestation_key_diversifiers.h"
+#include "sw/device/silicon_creator/lib/cert/cdi_0.h"  // Generated.
+#include "sw/device/silicon_creator/lib/cert/cdi_1.h"  // Generated.
+#include "sw/device/silicon_creator/lib/cert/uds.h"    // Generated.
+#include "sw/device/silicon_creator/lib/drivers/hmac.h"
+#include "sw/device/silicon_creator/lib/drivers/keymgr.h"
+#include "sw/device/silicon_creator/lib/drivers/lifecycle.h"
+#include "sw/device/silicon_creator/lib/otbn_boot_services.h"
+
+static keymgr_binding_value_t attestation_binding_value = {.data = {0}};
+static keymgr_binding_value_t sealing_binding_value = {.data = {0}};
+static attestation_public_key_t curr_pubkey = {.x = {0}, .y = {0}};
+static uint8_t cdi_0_tbs_buffer[kCdi0MaxTbsSizeBytes];
+static cdi_0_sig_values_t cdi_0_cert_params = {
+    .tbs = cdi_0_tbs_buffer,
+    .tbs_size = kCdi0MaxTbsSizeBytes,
+};
+static uint8_t cdi_1_tbs_buffer[kCdi1MaxTbsSizeBytes];
+static cdi_1_sig_values_t cdi_1_cert_params = {
+    .tbs = cdi_1_tbs_buffer,
+    .tbs_size = kCdi1MaxTbsSizeBytes,
+};
+
+static_assert(kAttestMeasurementSizeInBytes == 32,
+              "The attestation measurement size should equal the size of the "
+              "keymgr binding registers.");
+
+/**
+ * Returns true if debug (JTAG) access is exposed in the current LC state.
+ */
+static bool is_debug_exposed(void) {
+  lifecycle_state_t lc_state = lifecycle_state_get();
+  if (lc_state == kLcStateProd || lc_state == kLcStateProdEnd) {
+    return false;
+  }
+  return true;
+}
+
+status_t gen_uds_keys_and_cert(manuf_cert_perso_data_in_t *perso_data_in,
```

Comment:
nit: Since this module is pretty much for DICE. Should we just call it `dice.c`?

Then this function would be renamed to `dice_uds_cert_build()`

Another option would be `dice_cert.c`: `dice_cert_uds_build()`.

wdyt?

Reply:
sgtm! I will include this in the follow up PR that moves the owner manifest measurement to the CDI_1 stage.

Reply:
Added in #21539.